### PR TITLE
Added GoLang 1.10

### LINKED ISSUE #35: Add Go 1.10
to fix https://github.com/prometheus/prometheus/pull/3856

### PR DESCRIPTION
**Pull request by jamtur01**
Fixes #35.

I took a stab at this.

### CODE REVIEW

File: 1.10/Makefile.COMMON [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+REPOSITORY    := quay.io/prometheus
+NAME          := golang-builder
+BRANCH        := $(shell git rev-parse --abbrev-ref HEAD)
+SUFFIX        ?= -$(subst /,-,$(BRANCH))
+VERSION       := 1.10
```
Comment (sdurrheimer) [inline 1.10/Makefile.COMMON]: 1.10.0 even if golang doesn't show the .0

File: README.md [hunk 2]
```diff
@@ -7,10 +7,14 @@
 
 Docker Builder Image for cross-building Golang Prometheus projects.
 
-- `latest`, `main`, `1.9-main`, `1.9.2-main` ([1.9/main/Dockerfile](1.9/main/Dockerfile))
-- `arm`, `1.9-arm`, `1.9.2-arm` ([1.9/arm/Dockerfile](1.9/arm/Dockerfile))
-- `powerpc`, `1.9-powerpc`, `1.9.2-powerpc` ([1.9/powerpc/Dockerfile](1.9/powerpc/Dockerfile))
-- `mips`, `1.9-mips`, `1.9.2-mips` ([1.9/mips/Dockerfile](1.9/mips/Dockerfile))
+- `latest`, `main`, `1.10-main`, ([1.10/main/Dockerfile](1.10/main/Dockerfile))
```
Comment (sdurrheimer) [inline README.md]:
```Makefile
`latest`, `main`, `1.10-main`, `1.10.0-main` ([1.10/main/Dockerfile](1.10/main/Dockerfile))
```

File: README.md [hunk 3]
```diff
@@ -7,10 +7,14 @@
...
+- `arm`, `1.10-arm`, ([1.10/arm/Dockerfile](1.10/arm/Dockerfile))
```
Comment (sdurrheimer) [inline README.md]:
```Makefile
`arm`, `1.10-arm`, `1.10.0-arm` ([1.10/arm/Dockerfile](1.10/arm/Dockerfile))
```

File: README.md [hunk 4]
```diff
@@ -7,10 +7,14 @@
...
+- `powerpc`, `1.10-powerpc`, ([1.10/powerpc/Dockerfile](1.10/powerpc/Dockerfile))
```
Comment (sdurrheimer) [inline README.md]:
```Makefile
`powerpc`, `1.10-powerpc`, `1.10.0-powerpc` ([1.10/powerpc/Dockerfile](1.10/powerpc/Dockerfile))
```

File: README.md [hunk 5]
```diff
@@ -7,10 +7,14 @@
...
+- `mips`, `1.10-mips`, ([1.10/mips/Dockerfile](1.10/mips/Dockerfile))
```
Comment (sdurrheimer) [inline README.md]:
```Makefile
`mips`, `1.10-mips`, `1.10.0-mips` ([1.10/mips/Dockerfile](1.10/mips/Dockerfile))
```

**Comment by jamtur01** [comment]
@sdurrheimer I'd be happy to bump 1.9 and remove 1.8 if you like?

**Comment by sdurrheimer** [comment]
@jamtur01 Thank you, but I'm already working on it. =)

**Comment by jamtur01** [comment]
@sdurrheimer No worries - hope I didn't tread on toes? I was looking at the CircleCI builds last week as I was curious what they were doing. :)

**Comment by krasi-georgiev** [comment]
Thanks for the quick fix ;)

**Pull request merged**